Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.BaseCollection.cs

```diff
@@ -0,0 +1,106 @@
+// CodeContracts
+// 
+// Copyright (c) Microsoft Corporation
+// 
+// All rights reserved. 
+// 
+// MIT License
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System.Collections;
+using System.Drawing;
+using System.Diagnostics.Contracts;
+using System.Windows.Forms.Layout;
+using System.ComponentModel;
+
+namespace System.Windows.Forms
+{
+    /// <summary>
+    /// Provides the base functionality for creating data-related collections in the <see cref="N:System.Windows.Forms"/> namespace.
+    /// </summary>
+    public class BaseCollection : MarshalByRefObject // ICollection, IEnumerable
+    {
+        /// <summary>
+        /// Gets the total number of elements in the collection.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The total number of elements in the collection.
+        /// </returns>
+        
+        public virtual int Count
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<int>() > 0);
```

Comment:
This postcondition means that the collection could not be empty... Is it correct? Or postcondition should be `>=0`?